### PR TITLE
feat: add auth integration & make polyfill optional

### PR DESCRIPTION
**Pull request by nowaythatworked**
This PR makes it possible to use the package as a astro integration ~~(WARING: Right now, its not optional, because routes allways get added)~~ Endpoints can also be created manually.
The Crypto polyfill can be controlled dynamically with the 'edge' property in the config.
Config can be used with the getConfig/setConfig funcions in './config'.

To try it out set following options in your astro.config.ts
```ts
import { defineConfig } from "astro/config";
import auth from "auth-astro/integration";
import GitHub from "@auth/core/providers/github";

// https://astro.build/config
export default defineConfig({
  output: "server",
  integrations: [
    auth({
      authOptions: {
        trustHost: true,
        secret: import.meta.env.AUTH_SECRET,
        providers: [
          //@ts-expect-error issue https://github.com/nextauthjs/next-auth/issues/6174
          GitHub({
            clientId: import.meta.env.GITHUB_CLIENT_ID,
            clientSecret: import.meta.env.GITHUB_CLIENT_SECRET,
          }),
        ],
      },
    }),
  ],
});
```

### CODE REVIEW

File: config.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    authOptions: AstroAuthConfig
+}
+
+export const getConfig = (): AstroAuthIntegrationConfig => globalThis.astroAuthIntegrationConfig
+
+export const setConfig = (config: AstroAuthIntegrationConfig) => globalThis.astroAuthIntegrationConfig = config
+
+declare global {
+    var astroAuthIntegrationConfig: AstroAuthIntegrationConfig
+}
```
[the remark this answers is not in the document]
Comment (nowaythatworked) [reply config.ts:15]: Good point, will change that

File: index.ts [hunk 2]
```diff
@@ -23,27 +23,36 @@
...
-import crypto from 'node:crypto'
+import { Auth } from "@auth/core";
+import type { AuthConfig, AuthAction, Session } from "@auth/core/types";
+import {
+  type Cookie,
+  parseString,
+  splitCookiesString,
+} from "set-cookie-parser";
+import { serialize } from "cookie";
+import crypto from "node:crypto";
```
[the remark this answers is not in the document]
Comment (nowaythatworked) [reply index.ts:34]: I´m still looking for an optimal solution here, i got another way working in dev but on vercel it still crashes

[the remark this answers is not in the document]
Comment (TheOtterlord) [reply index.ts:34]: It's running node in dev afaik, but building the project should show any errors

**Comment by danielweck** [comment]
Hopefully this effort will supersede this one which seems stalled (albeit quite feature-rich):
https://github.com/astro-community/astro-auth

File: README.md [hunk 3]
```diff
@@ -1,51 +1,53 @@
...
+# Installation
 
-Install the core Auth.js package as well as the auth-astro wrapper.
-
-**info:** The auth-astro wrapper will not work independently, it relies on @auth/core as a dependency.
+The easiest way to get started is adding this package using the astro cli. 
 
 ```bash
-npm install auth-astro@latest @auth/core@latest
+yarn astro add auth-astro
```
Comment (TheOtterlord) [inline README.md:12]: any reason for specifing yarn? since npm is the default for most instructions I'd lean towards that

**Review by TheOtterlord — CHANGES_REQUESTED** [review]
Looks great! I've left some comments, but apart from those I believe we're ready to ship this thing :tada:

File: README.md (hunk 3 above)
Comment (nowaythatworked) [reply README.md:12]: No specific reason. I usally use yarn so it was in my head :D will change it 👍🏼

**Review by TheOtterlord — APPROVED** [review]

**Pull request merged**